omero/examples/java/JSONClient.java

```diff
@@ -0,0 +1,484 @@
+        List<JsonObject> result = new ArrayList<JsonObject>();
+        String url = serviceURLs.get("url:datasets");
+        JsonObject json = (JsonObject) get(url);
+        JsonArray data = json.getJsonArray("data");
+        for (int i = 0; i < data.size(); i++) {
+            result.add(data.getJsonObject(i));
+        }
+        return result;
+    }
+
+    /**
+     * List the images of a certain dataset
+     * 
+     * @param datasetId
+     *            The dataset id
+     * @return See above
+     * @throws Exception
+     *             If something went wrong
+     */
+    public Collection<JsonObject> listImages(int datasetId) throws Exception {
+        List<JsonObject> result = new ArrayList<JsonObject>();
+        Map<String, String> params = new HashMap<String, String>();
+        params.put("dataset", "" + datasetId);
+        String url = serviceURLs.get("url:images");
+        JsonObject json = (JsonObject) get(url, params);
+        JsonArray data = json.getJsonArray("data");
+        for (int i = 0; i < data.size(); i++) {
+            result.add(data.getJsonObject(i));
+        }
+        return result;
+    }
+
+    /**
+     * Update an object
+     * 
+     * @param object
+     *            The JSON object
+     * @return The updated object
+     * @throws Exception
+     *             If something went wrong
+     */
+    public JsonObject update(JsonObject object) throws Exception {
+        String url = serviceURLs.get("url:save");
+        JsonObject updatedObject = (JsonObject) put(url, object);
+        return updatedObject;
+    }
+
+    /**
+     * Perform a get request
+     * 
+     * @param urlString
+     *            The request URL
+     * @return The response
+     * @throws Exception
+     *             Exception If something went wrong
+     */
+    private JsonStructure get(String urlString) throws Exception {
+        return get(urlString, null);
+    }
+
+    /**
+     * Perform a GET request
+     * 
+     * @param urlString
+     *            The request URL
+     * @param params
+     *            The parameters
+     * @return The response
+     * @throws Exception
+     *             If something went wrong
+     */
+    private JsonStructure get(String urlString, Map<String, String> params)
+            throws Exception {
+        JsonReader reader = null;
+        try {
+            HttpGet httpGet = null;
+            if (params == null || params.isEmpty())
+                httpGet = new HttpGet(urlString);
+            else {
+                URIBuilder builder = new URIBuilder(urlString);
+                for (Entry<String, String> e : params.entrySet()) {
+                    builder.addParameter(e.getKey(), e.getValue());
+                }
+                httpGet = new HttpGet(builder.build());
+            }
+            HttpResponse res = httpClient.execute(httpGet);
+            reader = Json.createReader(new BufferedReader(
```

Comment:
I fixed the remarked issues in https://github.com/will-moore/ome-documentation/pull/4  Leave it to @will-moore to merge the changes in.